Update logging document to fix the inactive links (#654)
harvester/docs: docs/logging/harvester-logging.md

## High-level Architecture

> The [Logging Operator](https://kube-logging.dev/docs/#overview) now powers both `Harvester` and `Rancher` as an in-house logging solution.

Reviewer: We need to update those links in v1.2 doc, right ?

Author: v1.2 may be EOL. Please @jillian-maroket confirm, should we update v1.2 part as well? thanks.

Reviewer 2: According to the [Support Matrix](https://www.suse.com/suse-harvester/support-matrix/all-supported-versions/harvester-v1-2-x/), the EOL date for v1.2.x is 08 September 2024.

@bk201 Please confirm so we can finalize the scope of this PR. I'll also add the `EOL` label to v1.2.0 in the header's version selector.

Reviewer 3: Let's just update the non-EOL doc, starting from v1.3

###### Output

> The fields present in the **Output** form will change depending on the `Output` chosen, in order to expose the fields present for each [output plugin](https://kube-logging.dev/docs/configuration/plugins/outputs/).

Reviewer 2: ```suggestion
The form shows the fields that are available for the selected [output](https://kube-logging.dev/docs/configuration/plugins/outputs/).
```

###### Output Buffer

> The `Output Buffer` editor allows you to describe how you want the output buffer to behave. You can find the documentation for the buffer fields [here](https://kube-logging.dev/docs/configuration/plugins/outputs/buffer/).

Reviewer 2: ```suggestion
The editor allows you to describe the preferred output buffer behavior using various [fields](https://kube-logging.dev/docs/configuration/plugins/outputs/buffer/).
```

###### Filters

> Filters allow you to transform, process, and mutate the logs. In the text edit, you will find descriptions of the supported filters, but for more information, you can visit the list of [supported filters](https://kube-logging.dev/docs/configuration/plugins/filters/).

Reviewer 2: ```suggestion
Filters allow you to transform, process, and mutate the logs. For more information, see the list of supported [filters](https://kube-logging.dev/docs/configuration/plugins/filters/).
```

##### Referencing Secrets

> There are 3 ways `Logging Operator` allows specifying secret values via yaml values.

Reviewer 2: ```suggestion
You can define secret values (in YAML format) using any of the following methods:
```

         key: <kubernetes-secret-key>
```

> For more information, you can find the related documentation [here](https://kube-logging.dev/docs/configuration/output/).

Reviewer: ```suggestion
For more information, you can find the related documentation [here](https://kube-logging.dev/docs/configuration/plugins/outputs/secret/).
```

> For more information, you can find the related documentation [here](https://kube-logging.dev/docs/configuration/plugins/outputs/secret/).

Reviewer 2: ```suggestion
For more information, see [Secret definition](https://kube-logging.dev/docs/configuration/plugins/outputs/secret/).
```
